hoomd/hpmc/pytest/test_pair_expanded_gaussian.py

```diff
@@ -0,0 +1,222 @@
+    ),
+    (
+        dict(epsilon=5.0, sigma=1.1, delta=1.0, r_cut=2.0),
+        'shift',
+        1.5,
+        eg(1.5, 5, 1.1, 1.0) - eg(2.0, 5, 1.1, 1.0),
+    ),
+    (
+        dict(epsilon=5.0, sigma=1.1, delta=1.0, r_cut=2.5, r_on=2.0),
+        'xplor',
+        1.5,
+        eg(1.5, 5.0, 1.1, 1.0) * xplor_factor(1.5, 2.0, 2.5),
+    ),
+    (
+        dict(epsilon=5.0, sigma=1.1, delta=1.0, r_cut=2.5, r_on=2.0),
+        'xplor',
+        2.3,
+        eg(2.3, 5, 1.1, 1.0) * xplor_factor(2.3, 2.0, 2.5),
+    ),
+    (
+        dict(epsilon=5.0, sigma=1.1, delta=1.0, r_cut=2.0, r_on=3),
+        'xplor',
+        1.5,
+        eg(1.5, 5, 1.1, 1.0) - eg(2, 5, 1.1, 1.0),
+    ),
+    (
+        dict(epsilon=1.0, sigma=1, delta=1.0, r_cut=3.0, r_on=4),
+        'xplor',
+        3.2,
+        0,
+    ),
+]
+
+
+@pytest.mark.parametrize('params, mode, d, expected_energy',
+                         expanded_gauss_test_parameters)
+@pytest.mark.cpu
+def test_energy(mc_simulation_factory, params, mode, d, expected_energy):
+    """Test that ExpandedGaussian computes the correct energies for 1 pair."""
+    expanded_gauss = hoomd.hpmc.pair.ExpandedGaussian(mode=mode)
+    expanded_gauss.params[('A', 'A')] = params
+
+    simulation = mc_simulation_factory(d=d)
+    simulation.operations.integrator.pair_potentials = [expanded_gauss]
+    simulation.run(0)
+
+    assert expanded_gauss.energy == pytest.approx(expected=expected_energy,
+                                                  rel=1e-5)
+
+
+@pytest.mark.cpu
+def test_multiple_pair_potentials(mc_simulation_factory):
+    """Test that energy operates correctly with multiple pair potentials."""
+    expanded_gauss_1 = hoomd.hpmc.pair.ExpandedGaussian()
+    expanded_gauss_1.params[('A', 'A')] = dict(epsilon=1.0,
+                                               sigma=1.0,
+                                               delta=1.0,
+                                               r_cut=2.5)
+
+    expanded_gauss_2 = hoomd.hpmc.pair.ExpandedGaussian()
+    expanded_gauss_2.params[('A', 'A')] = dict(epsilon=2.0,
+                                               sigma=1.0,
+                                               delta=1.0,
+                                               r_cut=2.5)
+
+    # Some parameters are validated only after attaching.
+    simulation = mc_simulation_factory(1.5)
+    simulation.operations.integrator.pair_potentials = [
+        expanded_gauss_1, expanded_gauss_2
+    ]
+    simulation.run(0)
+
+    assert expanded_gauss_1.energy == pytest.approx(expected=eg(
+        1.5, 1.0, 1.0, 1.0),
+                                                    rel=1e-5)
+    assert expanded_gauss_2.energy == pytest.approx(expected=eg(
+        1.5, 2.0, 1.0, 1.0),
+                                                    rel=1e-5)
+    assert simulation.operations.integrator.pair_energy == pytest.approx(
+        expected=eg(1.5, 1.0, 1.0, 1.0) + eg(1.5, 2.0, 1.0, 1.0), rel=1e-5)
```

Comment:
You could save the expected results into variables. That would help with readability, especially the last line, which becomes
```python
assert simulation.operations.integrator.pair_energy == pytest.approx(expected_1 + expected_2)
```

Reply:
Apologies, I just noticed this feedback. I've just implemented your suggested changes and pushed to this branch.